src/fixing/Autofix.ml

```diff
@@ -181,6 +229,9 @@ let ast_based_fix ~fix (start, end_) (pm : Pattern_match.t) : Textedit.t option
         Autofix_printer.print_ast ~lang ~metavars ~target_contents
           ~fix_pattern_ast ~fix_pattern fixed_pattern_ast
       in
+      let text =
+        align_text_at_column ~start_column:(fst pm.range_loc).pos.column text
```

Comment:
I don't think this is right:

1. Won't this lead to double-indenting of the first line? I see that you've added tests, but there are quite a few different code paths that autofix can go through given the complexities of osemgrep vs pysemgrep and AST-based vs text-based autofix, so I'm worried that something may not be getting exercised here.
2. I think this will also cause issues if the autofix is applied partway through a line. There seems to be an implicit assumption here that an entire line is matched.

Reply:
This may not be clear from the name of the function, but `align_text_at_column` actually ignores the first line. I'll give it a more descriptive name / comment.

For the second point, I think this would actually depend on what our desired behavior is.

For instance, suppose we match `bar` in the line:
```
foo.bar.x
```
and we replace with the code
```
a()
b()
```
Now, this is already a pretty weird thing to do, but do we care if the result is:
```
foo.a()
b().x
```
or
```
foo.a()
    b().x
```

I believe that the current behavior would produce the former, and my change would produce the latter.

I can't think of a use case where this matters, but maybe you can.

Reply:
Yeah, I guess it's kind of a weird thing to do either way. I'm okay with the behavior you propose.